src/gsy_e/models/strategy/energy_parameters/energy_params_eb.py

```diff
@@ -21,37 +21,190 @@
+                the entire period of time over which the trade profile will be generated.
+            product_type: One of the available market types.
+        """
+
+    @property
+    @abstractmethod
+    def state(self) -> "StateInterface":
+        """Retrieve the internal state of the asset."""
+
+    @staticmethod
+    def day_forward_slots(market_slot: pendulum.DateTime):
+        """Get the market slots for the day forward market."""
+        return [
+            market_slot,
+            market_slot + pendulum.duration(minutes=15),
+            market_slot + pendulum.duration(minutes=30),
+            market_slot + pendulum.duration(minutes=45)]
+
+
+class ConsumptionStandardProfileEnergyParameters(ForwardEnergyParams):
     """Energy parameters of the load strategy for the forward markets."""
 
     def __init__(self, capacity_kW):
+        super().__init__()
         self.capacity_kW: float = capacity_kW
 
         self._state = LoadState()
         self._area = None
         self._profile_generator: Optional[ForwardTradeProfileGenerator] = None
 
+    @property
+    def state(self) -> LoadState:
+        """Retrieve the internal state of the asset."""
+        return self._state
+
     def serialize(self):
         """Return dict with the current energy parameter values."""
         return {
             "capacity_kW": self.capacity_kW
         }
 
+    @property
+    def capacity_kWh(self):
+        """Capacity of the load in kWh."""
+        return convert_kW_to_kWh(self.capacity_kW, self._area.config.slot_length)
+
+    def get_available_energy_kWh(
+            self, market_slot: pendulum.DateTime, market_type: AvailableMarketTypes):
+        """Get the available bid energy of the load."""
+        if market_type == AvailableMarketTypes.INTRADAY:
+            return self._state.get_energy_requirement_Wh(market_slot) / 1000.0
+        if market_type == AvailableMarketTypes.DAY_FORWARD:
+            return min(
+                self._state.get_energy_requirement_Wh(slot) / 1000.0
+                for slot in self.day_forward_slots(market_slot)
+            )
+
+        reference_slot = market_slot.set(hour=12, minute=0, tz=pendulum.UTC)
+
+        if self._state.get_desired_energy_Wh(reference_slot) <= FLOATING_POINT_TOLERANCE:
+            scaling_factor = 0.
+        else:
+            scaling_factor = (
+                    self._state.get_energy_requirement_Wh(reference_slot) /
+                    self._state.get_desired_energy_Wh(reference_slot))
+        return abs(scaling_factor) * self.capacity_kWh
+
     def event_activate_energy(self, area):
         """Initialize values that are required to compute the energy values of the asset."""
         self._area = area
         self._profile_generator = ForwardTradeProfileGenerator(
-            peak_kWh=convert_kW_to_kWh(self.capacity_kW, self._area.config.slot_length))
+            peak_kWh=self.capacity_kWh)
+        for i in range(4):
```

Comment:
This `event_activate_energy()` method doesn't distinguish between intraday/daily/etc. market types. How does this work in the case of an intraday market?

Reply:
This method is responsible for setting up the production / consumption forecast profile of an asset. It does not distinguish different markets, since the energy values that can be provided at each time period is constant and does not depend on the market type. Therefore it has to be done only once for the yearly market, since this is the market with the longest duration, in order to accomodate it. The other markets will operate on the same slots that were created for the yearly market. 
Having said that, I noticed that 4 years will not be enough since the yearly market spans across 6 years in the future. I'll change it with a constant that will control how far in the future can the forward markets reach.